### PR TITLE
Add custom build type, custom keystore path and adjust for custom generated applicationIDs

### PR DESCRIPTION
**Pull request by pavlospt**
Finally this one works... I have resorted to cherry-picking each commit and applying it after your upstream master latest changes.. I know that commit names are strange, but I want to make sure it passes the CI first and then I can squash them into one commit.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,6 +40,8 @@ Options
     -w, --wipe                       Wipe the temporary dryrun folder
     -h, --help                       Displays help
     -v, --version                    Displays the version
+    -k, --keystore KEYSTORE          Custom keystore path
+    --build_type BUILD_TYPE      Custom buildType (e.g. srv1, srv2)
```
Comment (cesarferreira) [inline README.md:44]: add a shortcut for the build type and ident according to the other items

File: dryrun.gemspec [hunk 2]
```diff
@@ -34,5 +37,6 @@ Gem::Specification.new do |spec|
   spec.add_dependency 'colorize', '~> 0.7'
   spec.add_dependency 'oga', '~> 1.3.1'
   spec.add_dependency 'highline', '~> 1.7'
+  spec.add_dependency 'adb-sdklib', '~> 0.0.3'
```
Comment (cesarferreira) [inline dryrun.gemspec:40]: why bring back this lib?

File: lib/dryrun.rb [hunk 3]
```diff
@@ -48,6 +52,14 @@ def create_options_parser(args)
           @flavour = flavour.capitalize
         end
 
+        opts.on('--build_type BUILD_TYPE', 'Custom buildType (e.g. srv1, srv2)') do |build_type|
```
Comment (cesarferreira) [inline lib/dryrun.rb:55]: add a shortcut

File: lib/dryrun.rb [hunk 4]
```diff
@@ -113,8 +125,6 @@ def pick_device
         @devices = DryrunUtils.run_adb('devices')
       end
 
-      puts 'No devices attached, but I\'ll run anyway' if @devices.empty?
```
Comment (cesarferreira) [inline lib/dryrun.rb:116]: why remove this? it's informative to the user

File: lib/dryrun/android_project.rb [hunk 5]
```diff
@@ -106,26 +129,20 @@ def install
...
 
-      remove_application_id
       remove_local_properties
 
-      if @custom_module
-        DryrunUtils.execute("#{builder} clean")
-        DryrunUtils.execute("#{builder} :#{@custom_module}:install#{@flavour}Debug")
-      else
-        DryrunUtils.execute("#{builder} clean")
+      root_module_builder = if @custom_module == nil
```
Comment (cesarferreira) [inline lib/dryrun/android_project.rb:134]: Please use a ternary operation

File: lib/dryrun/android_project.rb [hunk 6]
```diff
@@ -146,6 +163,9 @@ def sample_project
         @path_to_sample = File.join(@base_path, "/#{@custom_module}")
         return @path_to_sample, get_execution_line_command(@path_to_sample)
       else
+
+        @root_module = 'app'  if @modules.first.first.gsub('/','') == 'app'
```
Comment (cesarferreira) [inline lib/dryrun/android_project.rb:167]: why do you assume the module will be called `app` ?

File: lib/dryrun/android_project.rb [hunk 7]
```diff
@@ -203,19 +225,19 @@ def launchable_activity
...
 
     def get_launcher_activity(doc)
       activities = doc.css('activity')
       activities.each do |child|
         intent_filter = child.css('intent-filter')
 
-        if !intent_filter.nil? && !intent_filter.empty?
-          return child.attr('android:name').value
-        end
+      if intent_filter != nil and intent_filter.length != 0
```
Comment (cesarferreira) [inline lib/dryrun/android_project.rb:236]: why did you replace the code with this `non ruby standard` that does the same

**Review by cesarferreira — CHANGES_REQUESTED** [review]

**Comment by cesarferreira** [comment]
Did you try to use the tool? It's not working on my side.

For instance:
If I run this on the `master` branch it works great

```
dryrun https://github.com/UFreedom/FloatingView
```

with your code it doesn't

**Comment by cesarferreira** [comment]
Everything alright @pavlospt ?

**Comment by pavlospt** [comment]
Hey @cesarferreira, I have just been very busy lately and did not find some spare time to fix the issues. If you can leave it open till I can resolve them, it would be great. Otherwise, I completely understand if you just want to close it.

**Comment by cesarferreira** [comment]
Take your time mate, just check if I haven't added the features you want to include meanwhile

**Comment by cesarferreira** [comment]
I'll close it and will re-open when needed

**Comment by pavlospt** [comment]
Yeap, thanks, unfortunately I did not find the time to deal with it!

**Pull request closed without merging**